engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -4731,7 +4732,8 @@ protected VirtualMachine retrieve() {
         }
         catch (CloudRuntimeException e){
             s_logger.info("Caught CloudRuntimeException, returning job failed");
-            return new Pair<JobInfo.Status, String>(JobInfo.Status.FAILED, null);
+            CloudRuntimeException ex = new CloudRuntimeException("Unable to start VM instance : " + e);
+            return new Pair<JobInfo.Status, String>(JobInfo.Status.FAILED, JobSerializerHelper.toObjectSerializedString(ex));
```

Comment:
Why not return `e` instead of creating a new `ex`?

Reply:
@rhtyd : Only intention to create a new CloudRuntimeException ex is that proper error text can be shown to user in the output of startVirtualMachine API. Following is the output, user will get after this fix : 

`2017-12-06 17:11:10,329 DEBUG [o.a.c.f.j.i.AsyncJobManagerImpl] (API-Job-Executor-3:ctx-a8a7ec4b job-355) (logid:92a01680) Complete async job-355, jobStatus: FAILED, resultCode: 530, result: org.apache.cloudstack.api.response.ExceptionResponse/null/{"uuidList":[],"errorcode":530,"**errortext":"Unable to start VM instance**`

Reply:
@rhtyd : If you are fine with the fix then, can you please merge this PR?